wavecrestweb/kenworthy-machine
# Set up the pre-commit hook

## Description
This PR sets up a pre-commit hook that runs ESLint and Prettier on all **.js, .jsx, .ts, .tsx** files. As the name suggests, the pre-commit hook runs after you execute `git commit` command.


## Related Tickets & Documents


## Acceptance Criteria


- [x] Install [Prettier](https://prettier.io/docs/en/install)
- [x] Install [Husky](https://github.com/typicode/husky)
- [x] Install [lint-staged](https://github.com/lint-staged/lint-staged)
- [x] Set up **.prettierrc** (it's an empty config file to let editors and other tools know you are using Prettier)
- [x] Set up **.lintstagedrc.js** to integrate **lint-staged** with **Next.js**
- [x] Set up a pre-commit hook with **Husky**
- [x] Add scripts to **package.json**

## What type of PR is this? (check all applicable)


|     | Type                       |
| --- | -------------------------- |
|     | :sparkles: New feature     |
|     | 🎨 Style                   |
|     | :hammer: Refactoring       |
|     | 🔥 Performance Improvements|
|     | :bug: Bug fix              |
|     | :100: Add tests            |
|     | :link: Update dependencies |
|     | :scroll: Docs              |


## Mobile & Desktop Screenshots/Recordings



## What I learned


## Testing steps
1. Please make sure to run `npm i` to install the new packages locally. 
NOTE: Feel free to install [Prettier extension](https://github.com/prettier/prettier-vscode) for VScode , and adjust settings to set it as a default formatter, enable formating on save, etc.


## Added to documentation?

|     | Type                       |
| --- | -------------------------- |
|     | 📜 README.md    |
|     | 🙅 no documentation needed    |


## What gif best describes this PR?
